src/plugins/query_enhancements/server/routes/query_assist/ppl/time_parser_utils.ts

```diff
@@ -0,0 +1,211 @@
+    'YYYY-MM-DD HH:mm:ss',
+    'YYYY-MM-DD HH:mm:ss.SSS',
+    'YYYY-MM-DD HH:mm',
+    'YYYY/MM/DD HH:mm:ss',
+    'YYYY/MM/DD HH:mm:ss.SSS',
+    'YYYY/MM/DD HH:mm',
+    'YYYY-MM-DD[T]HH:mm:ss',
+    'YYYY-MM-DD[T]HH:mm:ss.SSS',
+    'YYYY-MM-DD[T]HH:mm',
+    'YYYY/MM/DD[T]HH:mm:ss',
+    'YYYY/MM/DD[T]HH:mm:ss.SSS',
+  ];
+
+  const m = moment(sanitizedTimeString, FORMATS, true);
+  if (!m.isValid()) {
+    return null;
+  }
+  return m;
+}
+
+/**
+ * Parses time range data from XML format
+ * Handles various edge cases and validates the time range values
+ *
+ * @param inputString - The string containing time range data (XML)
+ * @param logger - Logger instance for error reporting
+ * @returns The parsed time range object with formatted time strings or null if parsing fails
+ */
+export function parseTimeRangeXML(
+  inputString: string,
+  logger: Logger
+): { start: string; end: string } | null {
+  if (!inputString) {
+    logger.debug('Empty time range input string provided');
+    return null;
+  }
+
+  try {
+    // Extract start tag content
+    const startTagMatch = inputString.match(/<start>(.*?)<\/start>/s);
+    const startValue = startTagMatch ? startTagMatch[1].trim() : null;
+
+    // Extract end tag content
+    const endTagMatch = inputString.match(/<end>(.*?)<\/end>/s);
+    const endValue = endTagMatch ? endTagMatch[1].trim() : null;
+
+    if (!startValue || !endValue) {
+      logger.warn('No start or end tags found in XML time range');
+      return null;
+    }
+
+    const startDate = normTimeString(startValue);
+    const endDate = normTimeString(endValue);
+
+    if (!startDate || !endDate) {
+      return null;
+    }
+
+    if (startDate.isAfter(endDate)) {
+      return null;
+    }
+
+    return {
+      start: startDate.format('YYYY-MM-DD HH:mm:ss'),
+      end: endDate.format('YYYY-MM-DD HH:mm:ss'),
+    };
+  } catch (error) {
+    logger.error(`Error parsing time range input: ${error}`);
+    return null;
+  }
+}
+
+/**
+ * Retrieves clusters of timestamp fields (date type and their aliases) from an index using the OpenSearch mapping API
+ *
+ * @param indexName - The name of the index to query
+ * @param client - The OpenSearch client instance
+ * @param logger - Logger instance for error reporting
+ * @returns Promise resolving to an array of clusters; each cluster is an array of strings where the first element is the original field name, followed by its alias(es).
+ * @throws Error if the API call fails or index name is empty
+ */
+export async function getTimestampFieldClusters(
+  indexName: string,
+  client: OpenSearchClient,
+  logger: Logger
+): Promise<string[][]> {
+  if (!indexName) {
+    const error = new Error('Empty index name provided');
+    logger.error(error.message);
+    throw error;
+  }
+
+  try {
+    // Call the mapping API to get all field details
+    const response = await client.transport.request({
+      method: 'GET',
+      path: `/${indexName}/_mapping/field/*`,
```

Comment:
```suggestion
      path: `/${encodeURIComponent(indexName)}/_mapping/field/*`,
```
Let's encode the indexName to avoid the case users pass in indexName: `/foo/indexName`.